cli/pkg/cmd/upgrade/upgrade.go

```diff
@@ -0,0 +1,110 @@
+package upgrade
+
+import (
+	"cli/pkg/cmd/config"
+	"cli/pkg/console"
+	"cli/pkg/helm"
+	"cli/pkg/kube"
+	"cli/pkg/workspace"
+	"fmt"
+
+	"github.com/airyhq/airy/infrastructure/lib/go/k8s/util"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var (
+	approve    bool
+	version    string
+	Version    string
+	UpgradeCmd = &cobra.Command{
+		Use:   "upgrade [workspace directory]",
+		Short: "Upgrades an instance of Airy Core",
+		Long:  `Upgrades an existing Airy Core instance, read from a workspace directory (default .)`,
+		Args:  cobra.MaximumNArgs(1),
+		Run:   upgrade,
+	}
+)
+
+func init() {
+	UpgradeCmd.Flags().StringVar(&version, "version", "", "Specify a version to upgrade to (optional).")
+	UpgradeCmd.Flags().BoolVar(&approve, "approve", false, "Upgrade automatically without asking for an approval (optional).")
+}
+
+func upgrade(cmd *cobra.Command, args []string) {
+	workspacePath := ""
+	if len(args) > 0 {
+		workspacePath = args[0]
+	}
+
+	dir := workspace.Init(workspacePath)
+	conf, err := dir.LoadAiryYaml()
+	if err != nil {
+		console.Exit("Error parsing configuration file: ", err)
+	}
+	kubeCtx := kube.Load()
+	clientset, err := kubeCtx.GetClientSet()
+	namespace := viper.GetString("namespace")
+	if err != nil {
+		console.Exit("could not find an installation of Airy Core. Get started here https://airy.co/docs/core/getting-started/installation/introduction")
+	}
+
+	if version == "" {
+		version = Version
+	}
+	oldVersion := conf.Kubernetes.AppImageTag
+
+	fmt.Println("CLI version: ", Version)
+	fmt.Println("Current Airy Core version: ", oldVersion)
+	fmt.Println("New Airy Core version: ", version)
+	fmt.Println()
+
+	if util.ConfirmToProceed() == false {
+		console.Exit("Upgrade aborted.")
+	}
+
+	fmt.Println("Upgrading the helm charts of Airy Core...")
+	helm := helm.New(clientset, version, namespace, dir.GetAiryYaml())
+	if cmErr := helm.UpsertAiryConfigMap(); cmErr != nil {
+		console.Exit("Unable to copy config file into Airy Core.")
+	}
+	if upgradeErr := helm.UpgradeCharts(); upgradeErr != nil {
+		fmt.Println("Upgrading the Helm Charts failed with err: ", upgradeErr)
+		fmt.Println("Attempting to roll back the upgrade...")
+		if rollBackErr := helm.RollBackUpgrade(oldVersion); rollBackErr != nil {
```

Comment:
The roll-back is initiated only if `helm upgrade` fails. For the other steps, I'm not sure if it is needed or wise to do that.